### PR TITLE
Pretty printing + tests

### PR DESCRIPTION
**Pull request by faineance**
Changes:
- change directory structure to what other language packages seem to have.
- generate basic parser tests from files in tests/spec ( which are now quite correctly failing )
- add pretty printer

todo:
- [ ] fix wrapping of params
- [ ] implement the rest of expr
- [ ] make trailing ')' on new line.
- [ ] fix indentation generally
- [x] get golden tests working for binary writer

### CODE REVIEW

File: src/Entry.hs [hunk 1]
```diff
@@ -2,29 +2,29 @@ module Entry(
...
-import Data.Char
-import Data.Serialize
-import qualified Data.ByteString as BS
-
-import System.Process
-import System.Environment
-
-import Text.Show.Pretty
+import           Binary
+import           Core                         (toCore)
```
Comment (sdiehl) [inline src/Entry.hs]: This is kind of a weird way to format imports.

**Comment by faineance** [comment]
Yeah sorry my editor seems to do it automatically, let me turn it off.

**Comment by sdiehl** [comment]
All looks like good progress. Can merge this into mainline whenever the tests are good to go.

**Comment by faineance** [comment]
Tests are now passing, going to squash and merge.

**Pull request merged**